test/support/test_config.h

```diff
@@ -154,6 +154,9 @@
 #define _PSTL_TEST_COMPLEX_SINH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
 #define _PSTL_TEST_COMPLEX_TANH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
 
+#define _PSTL_TEST_COMPARISON_BROKEN                                                                                   \

```

Comment:
Since we are defining this macro and are not first checking if `_MSVC_LANG` and other macros are present, then I believe this will cause a compile error on Linux due to the following usages:

`return TestUtils::done(!_PSTL_TEST_COMPARISON_BROKEN);`

Reply:
Good point.  Why did this not trip our per commit CI...?

Reply:
Ah, its a blind spot for our per commit CI because only the sycl_iterator tests run for dpcpp backend.
I'll make a fix.  Thanks.